templates/workspace_services/azureml/terraform/network.tf

```diff
@@ -0,0 +1,62 @@
+
+data "azurerm_network_security_group" "ws" {
+  name                = "nsg-ws"
+  resource_group_name = data.azurerm_resource_group.ws.name
+}
+
+
+resource "azurerm_network_security_rule" "allow-BatchNodeManagement-inbound-29877" {
+  access                      = "Allow"
+  destination_port_range      = "29877"
+  destination_address_prefix  = "VirtualNetwork"
+  source_address_prefix       = "BatchNodeManagement"
+  direction                   = "Inbound"
+  name                        = "BatchNodeManagement-inbound-29877"
+  network_security_group_name = data.azurerm_network_security_group.ws.name
+  priority                    = 200
+  protocol                    = "TCP"
+  resource_group_name         = data.azurerm_resource_group.ws.name
+  source_port_range           = "*"
+}
+
+resource "azurerm_network_security_rule" "allow-BatchNodeManagement-inbound-29876" {
+  access                      = "Allow"
+  destination_port_range      = "29876"
+  destination_address_prefix  = "VirtualNetwork"
+  source_address_prefix       = "BatchNodeManagement"
+  direction                   = "Inbound"
+  name                        = "BatchNodeManagement-inbound-29876"
+  network_security_group_name = data.azurerm_network_security_group.ws.name
+  priority                    = 201
+  protocol                    = "TCP"
+  resource_group_name         = data.azurerm_resource_group.ws.name
+  source_port_range           = "*"
+}
+
+resource "azurerm_network_security_rule" "allow-AzureMachineLearning-inbound-44224" {
+  access                      = "Allow"
+  destination_port_range      = "44224"
+  destination_address_prefix  = "VirtualNetwork"
+  source_address_prefix       = "*"
+  direction                   = "Inbound"
+  name                        = "AzureMachineLearning-inbound-44224"
+  network_security_group_name = data.azurerm_network_security_group.ws.name
+  priority                    = 202
+  protocol                    = "TCP"
+  resource_group_name         = data.azurerm_resource_group.ws.name
+  source_port_range           = "*"
+}
+
+resource "azurerm_network_security_rule" "allow-Outbound_Storage_445" {
```

Comment:
Does this risk data being extracted from the VNet by users to other storage accounts?

Reply:
Yes, we'll need to clarify with PG why this is needed, but it is checked by compute cluster that it has to be there (compute can't be created without it). Max limitation is region.